src/ctask.jl

```diff
@@ -203,7 +209,7 @@ function consume(ctask::CTask, values...)
         push!(producer.storage[:consumers].waitq, ct)
     end
 
-    if getproperty(producer, :state) === :runnable
+    if producer.state === :runnable
```

Comment:
Maybe also here use something that is less dependent on the internal state?
```suggestion
    if !istaskdone(producer) && !istaskfailed(producer)
```

Reply:
There's another potential state (`:queued`) here, so I think using `=== :runnable` is more precise here.

Reply:
I think it was removed already before Julia 1.3, so it should not matter anymore.

Reply:
I couldn't find it in https://github.com/JuliaLang/julia/blob/v1.3.0-alpha/base/task.jl.

Reply:
Another indication is this check in ctask.jl: https://github.com/TuringLang/Libtask.jl/blob/a664d3dc5ee8e56ab85dc9368e94f68c8b150083/src/ctask.jl#L144-L146 Apparently it was a thing only in Julia < 1.2.

Reply:
You are right, updated.